processor/k8sattributesprocessor/pod_association.go

```diff
@@ -99,6 +100,19 @@ func getConnectionIP(ctx context.Context) kube.PodIdentifier {
 	case *net.IPAddr:
 		return kube.PodIdentifier(addr.IP.String())
 	}
+
+	//If this is not a known address type, check for known "untyped" formats.
+	// 1.1.1.1:<port>
+
+	splitIPAndPort := strings.Split(c.Addr.String(), ":")
+	if len(splitIPAndPort) > 1 {
+		ipString := strings.Join(splitIPAndPort[:len(splitIPAndPort)-1], ":")
```

Comment:
This looks like it could be more readable (and a bit more performant probably) using `strings.LastIndexByte` and a slice on the original string, rather than split then join

Reply:
great suggestion! switched to using strings.LastIndex